js/runner.go

```diff
@@ -110,6 +111,38 @@ func (r *Runner) NewVU(samplesOut chan<- stats.SampleContainer) (lib.VU, error)
 	return lib.VU(vu), nil
 }
 
+func rr_getAddr(ifaceName string) *net.IPAddr {
+	//rand.Seed(time.Now().Unix())
+
+	// Select given interface
+	targetIf, err := net.InterfaceByName(ifaceName)
+	if err != nil {
+		return nil
+	}
+	// Get available addresses for interface
+	addrs, err := targetIf.Addrs()
+	if err != nil {
+		return nil
+	}
+	rand.Shuffle(len(addrs), func(i, j int) { addrs[i], addrs[j] = addrs[j], addrs[i] })
+	ip := ""
+	for _, a := range addrs {
+		tip, _, err := net.ParseCIDR(a.String())
+		if err == nil && tip.String() != "::1" {
```

Comment:
You specifically fiter out the IPv6 loopback address, but not the IPv4 ones? Not sure what the correct behavior should be here, just mentioning the inconsistency